### ISSUE TITLE
Type narrowing with metaclasses doesn't work in some situations

### ISSUE THREAD

**Issue by Aran-Fey** [microsoft/pyright#7008]
```python
class Meta(type):
    foo: str

    def __init__(cls, *_):
        for base_cls in cls.__bases__:
            if isinstance(base_cls, Meta):
                reveal_type(base_cls.foo)  # Unknown
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.348, which was just published. It will also be included in a future release of pylance.
